mllib/src/main/scala/org/apache/spark/mllib/linalg/CholeskyDecomposition.scala

```diff
@@ -52,8 +51,20 @@ private[spark] object CholeskyDecomposition {
   def inverse(UAi: Array[Double], k: Int): Array[Double] = {
     val info = new intW(0)
     lapack.dpptri("U", k, UAi, info)
-    val code = info.`val`
-    assert(code == 0, s"lapack.dpptri returned $code.")
+    checkReturnValue(info, "dpptri")
     UAi
   }
+
+  private def checkReturnValue(info: intW, method: String): Unit = {
+    info.`val` match {
+      case code if code < 0 =>
+        throw new IllegalStateException(s"LAPACK.$method returned $code; arg ${-code} is illegal")
```

Comment:
Yes, none of this info is really useful to the end user in this case because it's a bug. It may be useful in an error report. I figured it's worth saying _something_ and why not say a little more about the cause to help anyone who cares save the step of looking up what "-2" means.